src/modules/file/file.controller.ts

```diff
@@ -54,12 +57,26 @@ export class FileController {
   ) {
     const file = await this.fileService.findByIdAndUserId(id, user.id);
 
+    const filesize = await this.fileService.getFilesize(data.body);
+
+    const bytesSize = await this.redisService.get(id);
+    const possibleBytesSize = bytesSize + filesize;
+    if (possibleBytesSize > user.f_size_max) {
+      throw new HttpException(
+        UserError.LimitExceeded,
+        HttpStatus.FORBIDDEN,
+      );
+    }
+
+    await this.redisService.incrBy(id, filesize);
     await this.fileService.save({
       ...file,
+      filesize: file.filesize + filesize,
```

Comment:
FIlesize in DB may be save upon finalizing object on S3. We don't have to touch DB on every chunk push.